File: SudokuSolver.java
/*
 * SudokuSolver.java
 * This sudoku solver will generate a solution to any valid sudoku puzzle
 *
 * @author Kevin Tone
 */



public class SudokuSolver() {
	
	private int boardArray[][];
	
		
	public SudokuSolver()	{
		//Initialize the board
		boardArray = new int[9][9];
	}
	
	public void getBoard() {
		
	}

	public void printBoard() {
		for (int i = 0; i < 9; i++) {
			for (int j = 0; j < 9; j++) {
				System.out.print(boardArray[i][j]);
			}
			System.out.println();
		}
	}

}

File: c/SudokuSolver.c



#include "SudokuSolver.h"

int[][] *boardArray = NULL;

int getBoard(int[ARRAY_LENGTH][ARRAY_LENGTH] board)
{
	*boardArray = malloc(sizeof(int) * 81);
	setBoardToNull();
}

void setBoardToNull()
{
	int i,j;
	for(i = 0; i < ARRAY_LENGTH + 1; i++)
	{
		for(j=0; j<ARRAY_LENGTH+1; j++)
		{
			boardArray[i][j] = NULL;
		}
	}
}

int printBoard() {
	int i, j;
	if(*boardArray != NULL)
	{
		for(i = 0; i < ARRAY_LENGTH+1; i++)
		{
			for(j = 0; j < ARRAY_LENGTH+1; j++)
			{
				if(boardArray[i][j] == NULL)
				{
					break;
				} else
				{
					printf("%i|", boardArray[i][j]);
				}
			}
			printf("\n");
		}
	}
}


void clearData()
{
	int i, j;
	for(i=0; i<ARRAY_LENGTH; i++)
	{
		for(j=0; j<ARRAY_LENGTH; j++)
		{
			free(boardArray[i][j]);
		}
	}
	free(boardArray);
}



